### PR TITLE
Fix TS errors and remove ts-ignore

### PR DESCRIPTION
**Pull request by PanRadaDev**
Samo improvements code to remove ts-ignore and fix errors

WIP!

### CODE REVIEW

File: src/components/InputItem.tsx [hunk 1]
```diff
@@ -45,7 +45,6 @@ export class InputItem extends PureComponent<Props, State> {
     this.setState({
       isActive: true,
     });
-    // @ts-ignore
     Animated.timing(this.state.isAnimatedFocused, {
```
Comment (Kamill90) [inline src/components/InputItem.tsx:48]: don't you need to specify `useNativeDriver` option?

File: src/screens/CreateContactScreen.tsx [hunk 2]
```diff
@@ -17,8 +17,11 @@ const i18n = require('../../loc');
 
 interface Props {
   navigation: CompositeNavigationProp<
-    StackNavigationProp<MainTabNavigatorParams, Route.ContactList>,
-    StackNavigationProp<MainCardStackNavigatorParams, Route.CreateContact>
+    StackNavigationProp<MainCardStackNavigatorParams, Route.CreateContact>,
+    CompositeNavigationProp<
+      StackNavigationProp<MainTabNavigatorParams, Route.ContactList>,
+      StackNavigationProp<MainCardStackNavigatorParams, Route.ScanQrCode>
+    >
```
Comment (Kamill90) [inline src/screens/CreateContactScreen.tsx:24]: Awesome, could you share the part of the manual that has brought you here?

Comment (PanRadaCloudBest) [reply src/screens/CreateContactScreen.tsx:24]: not sure where exact but simple browse react-navigation API as remember

Comment (PanRadaCloudBest) [reply src/screens/CreateContactScreen.tsx:24]: any way need implementing to others similar case

File: src/components/Header.tsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ export const HEADER_HEIGHT = Platform.select({
 }) as number;
 
 interface Props {
-  navigation?: StackNavigationProp<any, any>;
+  navigation?: any; // TODO
```
Comment (KedziaPawel) [inline src/components/Header.tsx:21]: seems that it doesn't make sense to merge it with `any` here as turn off checking types completely for navigation. If it needs a refactor to make it work I would make it and merge it after a hard fork

Comment (PanRadaCloudBest) [reply src/components/Header.tsx:21]: yes for me to it is not resolve

File: src/services/NavigationService.ts [hunk 4]
```diff
@@ -1,10 +1,18 @@
...
+export const navigationRef = createRef<NavigationContainerRef>();
 
 export default class NavigationService {
-  navigate = (name: string, params: unknown) => {
-    navigationRef.current?.navigate(name, params);
-  };
+  goBack() {
+    navigationRef.current?.goBack();
+  }
+  navigate(routeName: string, params?: Record<string, any>) {
```
Comment (KedziaPawel) [inline src/services/NavigationService.ts:10]: Zrzut ekranu 2020-11-20 o 11 16 31
with this we will have routeName and params typescript validation.

Remove: Unlock, DeleteWallet, ReleaseNotes from Routes enum. They are not unused, someone had to forgot to delete them when deleting screen

File: src/services/NavigationService.ts [hunk 5]
```diff
@@ -1,10 +1,20 @@
...
+export const navigationRef = createRef<NavigationContainerRef>();
 
 export default class NavigationService {
-  navigate = (name: string, params: unknown) => {
-    navigationRef.current?.navigate(name, params);
-  };
+  goBack() {
+    navigationRef.current?.goBack();
+  }
+  navigate(routeName: string, params?: GlobalParams[Route]) {
```
Comment (KedziaPawel) [inline src/services/NavigationService.ts:12]: routeName: Route

Comment (Kamill90) [reply src/services/NavigationService.ts]: done

**Review by KedziaPawel — APPROVED** [review]

**Pull request merged**